service-registry/src/main/java/org/apache/servicecomb/serviceregistry/api/registry/MicroserviceFactory.java

```diff
@@ -53,8 +54,13 @@ private Microservice createMicroserviceFromDefinition(Configuration configuratio
     microservice.setServiceName(configuration.getString(CONFIG_QUALIFIED_MICROSERVICE_NAME_KEY,
         DEFAULT_MICROSERVICE_NAME));
     microservice.setAppId(configuration.getString(CONFIG_APPLICATION_ID_KEY, DEFAULT_APPLICATION_ID));
-    microservice.setVersion(configuration.getString(CONFIG_QUALIFIED_MICROSERVICE_VERSION_KEY,
-        DEFAULT_MICROSERVICE_VERSION));
+    String version = configuration.getString(CONFIG_QUALIFIED_MICROSERVICE_VERSION_KEY,
+        DEFAULT_MICROSERVICE_VERSION);
+    //check version format
+    if (!Version.checkVersion(version)) {
```

Comment:
why not just new a Version instance

Reply:
Maybe just check format is better ? the new Version instance will never used

Reply:
but constrcut a version will got the detail of the wrong format message

Reply:
do you want to express version format check logic by two different way?

Reply:
ok, I see.